### DISCUSSION TITLE
which code  should I upload?

### DISCUSSION THREAD

**Discussion by souravlal** [marcel-licence/esp32_midi_sampler#23] [Q&A] [answered]
In this project I can see a few program files (.ino), which one should I upload it to my Esp32 audio dev- kit? Should I click Littlefs data upload? if so which are the files need to include in to data folder?
I am not new to Arduino but new to the Esp32 please guide me through.

**Comment by marcel-licence** [comment]
Using ESP32 LittleFS Data Upload will automatically upload files in the data directory. This is recommended to allow the firmware to autoload my example sample I've provided.
ESP32 Sketch Data Upload should compile and upload the firmware. Ensure that PSRAM is enabled. (The arrow right button is also for compile and upload).
The file to open is always the one which has the same name as the containing folder (esp32_midi_sampler.ino). Other files are found automatically by arduino and also compiled and linked together.
Hope this helps

**Reply by souravlal** [reply]
Do I need to download the zip file with all the files?
Where should I place it in my computer?
esp32_midi_sampler.ino is the code to upload?

**Comment by Slider2732** [comment]
I may be able to help too.
Yep, unzip the zip and put it wherever makes sense. I have a folder called 'ML Synth Projects' where his projects go on my Desktop. Others may put the folder in the Sketchbook folder of the Arduino IDE. Wherever you'd normally put project files you may be working on, the IDE doesn't care. It will normally want to save files in the Sketchbook folder, but you can pick anywhere.
 
Open esp32_midi_sampler.ino. It needs the others to run, such as the MIDI section, VU meter, sampler section etc. All the individual parts make such projects easier for identifying the functions they have within the whole sampler project. They will run across the top line of the IDE as their file names from the zip.
When the IDE uploads to the ESP32, it assembles the various .ino's of the project into the full code that will run. 

It's a bit like if you wrote a sketch for a smart watch. You might have separate bits of code for watch face designs, internet functions, screen display, power management. Marcel has put what could have been a huge single sketch into smaller sections.  

Also be sure to downgrade the ESP32 software version. Something doesn't work with the latest release. 
Tools --> Boards Manager
Then type 'ESP32'
The version should be 1.0.4 not 1.0.6

**Reply by jazzbiddy** [reply]
@Slider2732 When you say 
> Something doesn't work with the latest release

Do you mean it doesn't compile, or did it compile and upload, but you ran into other issues?

I'm just getting started with this and was able to get it to compile and upload using 1.0.6 - but now I'm wondering if I'm going to have some other issues. 
thanks

**Reply by marcel-licence** [reply]
I moved also over to 1.0.6 but not all projects are updated yet

**Comment by Slider2732** [comment, marked as answer]
It's only as far as I know there are issues for some projects. I know I uploaded with 1.0.6 and got no sound, re-uploaded with 1.0.4 and got sound, but I tried changing a few other things in that time so my data isn't valid.  
Not got a clue of the specifics, all I know is that all of Marcel's projects work with 1.0.4 
The safest bet to know a project will work. 
It could be function timings or anything, I changed to 1.0.4 and forgot about it  :)

If it has uploaded fine with the newer version, go for it and hopefully it works. It would be cool to know either way. His code has changed with new additions like the analog key inputs support, so it's worth plugging stuff in and seeing.

**Reply by jazzbiddy** [reply]
Thanks for the extra info. I'll give it a shot. And now I know what the issue might be if I run into issues.

**Comment by souravlal** [comment]
Do I need to hit the 'esp32 sketch upload' option or upload the code the normal way?

**Reply by marcel-licence** [reply]
@souravlal If you mean by normal way hitting the right arrow button then both should have the same functionality. =)

**Reply by souravlal** [reply]
'Esp32 sketch data upload' or 'ESP32 littlefs data upload'?
Which should I click to upload samples to esp32?
